### PR TITLE
SuppliesView: Gtk4 Prep

### PR DESCRIPTION
**Pull request by danirabbit**
Screenshot from 2023-12-20 12 42 33

Printer:
* Can't set `default` to false, so use a function here for clearer API

PrinterPage:
* Add set default button here. Can't set false so switch doesn't make sense.

SuppliesView:
* Don't nest a scrolledwindow. This whole view is already inside a scrolledwindow. Subclass box instead
* Use GraniteHeaderLabel and set mnemonic widgets for a11y
* Code style
* Bind from the source of truth, not from the widget. Use SYNC_CREATE saves setting default values manually
* Combine InkLevel widget since we can't subclass FlowBox in GTK 4 and these are small classes

### CODE REVIEW

**Comment by danirabbit** [comment]
I added a snippet to the Wiki we can use to test ink levels for those of us without a printer https://github.com/elementary/switchboard-plug-printers/wiki/Test-Ink-Levels

**Comment by danirabbit** [comment]
@jeremypw hoping to fix the issue you were experiencing in #164 here

File: src/Views/SuppliesView.vala [hunk 1]
```diff
@@ -1,72 +1,141 @@
...
     construct {
-        var name_label = new Gtk.Label (_("Description:"));
-        name_label.xalign = 1;
+        var name_entry = new Gtk.Entry () {
+            hexpand = true,
+            placeholder_text = _("BrandPrinter X3000")
+        };
+
+        var name_label = new Granite.HeaderLabel (_("Description")) {
+            mnemonic_widget = name_entry
```
Comment (jeremypw) [inline src/Views/SuppliesView.vala:29]: Currently has no effect? I presume a mnemonic will be set after porting to Granite-7.

**Review by jeremypw — APPROVED** [review]
All seems to work as expected and code looks good. 

The test snippet could be improved by using a range of level, max_level, min_level and maybe suggestion where to insert into the code.

One curious thing I noticed when changing the test values was that unless the level was `0`, the ink box showed some ink was present even if the level was below the minimum level. It also shows ink if the level is negative. Is this a Gtk issue? It can be worked around but as the printer minimum level is presumably not intended to indicate no ink but the minimum recommended fill level, it would be nice to show this on the scale somehow.  However, that can be left to another PR.

**Pull request merged**